### ISSUE TITLE
Unicode support

### ISSUE THREAD

**Issue by bkazemi** [scanmem/scanmem#211]
`libscanmem` should handle Unicode strings. I'll add this to my queue.

**Comment by 12345ieee** [comment]
Technically they are supported now, as searching for `☺à☺à☺` gives the correct results, but it's more thanks to C99's magic than `scanmem` itself.

I hope for your sanity you only plan to implement UTF-8.

@sriemer : Proposed for milestone v0.17 label 'enhancement'.

**Comment by bkazemi** [comment]
Yeah, this issue will be more about implementing a safe system to avoid nasty charset issues. I'll edit the title to avoid some ambiguity.

`libscanmem` should probably just support UTF-8. It's pretty much the de facto encoding for unicode these days.

**Comment by sriemer** [comment]
Many tools also use UTF-16. An example is `khangman`. Unicode strings are an enhancement and shouldn't be part of `v0.16`.
